docs/frameworks/retries.rst

```diff
@@ -0,0 +1,28 @@
+.. _framework-retries:
+
+==================
+Supporting Retries
+==================
+
+Web-poet frameworks must catch :exc:`~web_poet.exceptions.core.Retry`
+exceptions raised from the :meth:`~web_poet.pages.ItemPage.to_item` method of a
+page object.
+
+When :exc:`~web_poet.exceptions.core.Retry` is caught:
+
+#.  The original request whose response was fed into the page object must be
+    retried.
+
+#.  A new page object must be created, of the same type as the original page
+    object, and with the same input, except for the response data, which must
+    be the new response.
```

Comment:
I'm not sure that's valid. All inputs which correspond to the same page should be re-created, there shouldn't be a scenario when one input comes from an original page, and the other one comes from a retried result.

Reply:
I imagine this is worded like this to account for inputs that should not change during a retry, which I assume is the case of RequestUrl and PageParams. We could reword it to talk about updating inputs that can be updated, although it may be best to be specific about what input should and which inputs should not be updated during a retry.